### PR TITLE
Adds support to sign the request using amazon lambda

### PR DESCRIPTION
**Pull request by mcfedr**
Its no longer necessary to run a server to do the signing requests, rather you can use the AWS Lambda service to host the function.

### CODE REVIEW

**Comment by kookster** [comment]
We use a lambda for this as well, but we fronted it using the Amazon API Gateway so that we did not need to change evaporate, or include the aws sdk js, to be able to get back signatures.

**Comment by mcfedr** [comment]
Interesting, thats something that didn't exist back when I made this, or I just didn't get as far as integrating it...

**Comment by bikeath1337** [comment]
@mcfedr , I brought your branch current with master and then renamed the options to be consistent with other named options; updated the library references a bit and added a bit more documentation.

Would you be able to review and give it a spin so we can consider this as a candidate to merge?

Thanks!

https://github.com/bikeath1337/EvaporateJS/commits/lambda

**Comment by mcfedr** [comment]
@bikeath1337 It looks good, ill give it a quick test this afternoon

**Comment by bikeath1337** [comment]
Hi, there, @mcfedr . Hope you had a great weekend.

Were you able to check this out? Thanks!

**Comment by mcfedr** [comment]
Hi, I have updated this pull request from yours, I have added one more commit, https://github.com/TTLabs/EvaporateJS/pull/78/commits/7434642d34d8989bc40c40fe68c395cb82365cdc that corrects a couple of small things
- Somehow the function `authorizedSignWithLambda` was added twice
- It missed how I had moved `encodeURIComponent`
- I renamed lambdaMethod as lambdaFunction as this better reflects the naming Amazon uses

**Comment by bikeath1337** [comment]
Thanks, I merged your changes and squashed all three commits on my `lambda` branch. Want to quickly check?

File: README.md [hunk 1]
```diff
@@ -202,6 +204,8 @@ The `.add()` method returns the internal EvaporateJS id of the upload to process
 
 * **signParams**: _Object_. an object of key/value pairs that will be passed to _all_ calls to the signerUrl. 
 
+* **signHeaders**: _Object_. an object of key/value pairs that will be passed as headers to _all_ calls to the signerUrl.
```
Comment (mcfedr) [inline README.md]: I noticed this was missing from the README file

File: evaporate.js [hunk 2]
```diff
@@ -1180,6 +1180,41 @@
...
+                        me.warn(warnMsg);
+                        authRequester.onFailedAuth(err);
+                        return;
+                    }
+                    authRequester.auth = JSON.parse(data.Payload);
+                    authRequester.onGotAuth();
+                });
+            }
+
+            function makeSignParamsObject(params) {
```
Comment (mcfedr) [inline evaporate.js]: Consolidated the logic to do this in one place

**Comment by mcfedr** [comment]
Sorry, just pushed one more commit, I noticed the sign params were not being sent to lambda, I think this is quite important

**Comment by mcfedr** [comment]
Squash and pushed here so its easier for you

**Comment by bikeath1337** [comment]
Thank you, Fred. I updates the Evaporate instantiation call to illustrate the two options that were previously hidden in the code and made one style adjustment to your changes.

If you can review, I would appreciate it. If all pans out, I will package this as a candidate release. @tomsaffell, what think ye?

**Comment by mcfedr** [comment]
Looks fine to me

**Comment by bikeath1337** [comment]
@mcfedr I published this to NPM as v1.1.0

**Comment by tomsaffell** [comment]
@bikeath1337 sorry I missed this. seems like a great idea. still need anything from me?

**Comment by bikeath1337** [comment]
@tomsaffell I think we're good!

**Pull request merged**